src/protocol/commands/_qThreadExtraInfo.rs

```diff
@@ -0,0 +1,31 @@
+use super::prelude::*;
+
+use crate::protocol::common::thread_id::ThreadId;
+use crate::protocol::SpecificThreadId;
+
+#[derive(Debug)]
+pub struct qThreadExtraInfo<'a> {
+    pub id: SpecificThreadId,
+
+    pub buf: &'a mut [u8],
+}
+
+impl<'a> ParseCommand<'a> for qThreadExtraInfo<'a> {
+    #[inline(always)]
+    fn from_packet(buf: PacketBuf<'a>) -> Option<Self> {
+        let (buf, body_range) = buf.into_raw_buf();
+        let (body, buf) = buf[body_range.start..].split_at_mut(body_range.end - body_range.start);
```

Comment:
take a look at how “vFilePread` is implemented. 

In this case, since you don't need to hold on to any references into the `buf` after parsing is complete, you are able to reuse the entirety of the buffer.

Reply:
Thanks for the pointer, I've matched the logic